R/utilities-data-set.R

```diff
@@ -55,38 +55,64 @@ saveDataSetToPKML <- function(dataSet, filePath) {
-#' xDimension, xUnit, yDimension, yUnit, yErrorType, yErrorUnit, yMolWeight
+#' @return DataSet objects as data.frame with columns name, xValues, yValues, yErrorValues,
+#' xDimension, xUnit, yDimension, yUnit, yErrorType, yErrorUnit, molWeight, lloq,
+#'  and a column for each meta data that is present in any `DataSet`
 #' @export
 dataSetToDataFrame <- function(dataSets) {
   dataSets <- c(dataSets)
   validateIsOfType(dataSets, DataSet)
 
-  name <- unlist(mapply(.makeDataFrameColumn, dataSets, "name"))
-  xUnit <- unlist(mapply(.makeDataFrameColumn, dataSets, "xUnit"))
-  yUnit <- unlist(mapply(.makeDataFrameColumn, dataSets, "yUnit"))
-  yErrorUnit <- as.character(unlist(mapply(.makeDataFrameColumn, dataSets, "yErrorUnit")))
-  xDimension <- unlist(mapply(.makeDataFrameColumn, dataSets, "xDimension"))
-  yDimension <- unlist(mapply(.makeDataFrameColumn, dataSets, "yDimension"))
-  yErrorType <- as.character(unlist(mapply(.makeDataFrameColumn, dataSets, "yErrorType")))
-  yMolWeight <- unlist(mapply(.makeDataFrameColumn, dataSets, "molWeight"))
-  xValue <- unlist(mapply(.makeDataFrameColumn, dataSets, "xValues"))
-  yValue <- unlist(mapply(.makeDataFrameColumn, dataSets, "yValues"))
-  yErrorValues <- unlist(mapply(.makeDataFrameColumn, dataSets, "yErrorValues"))
+  name <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "name")
+  }), use.names = FALSE)
+  xUnit <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "xUnit")
+  }), use.names = FALSE)
+  yUnit <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "yUnit")
+  }), use.names = FALSE)
+  yErrorUnit <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "yErrorUnit")
+  }), use.names = FALSE)
+  xDimension <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "xDimension")
+  }), use.names = FALSE)
+  yDimension <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "yDimension")
+  }), use.names = FALSE)
+  yErrorType <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "yErrorType")
+  }), use.names = FALSE)
+  molWeight <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "molWeight")
+  }), use.names = FALSE)
+  xValues <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "xValues")
+  }), use.names = FALSE)
+  yValues <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "yValues")
+  }), use.names = FALSE)
+  yErrorValues <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "yErrorValues")
+  }), use.names = FALSE)
+  lloq <- unlist(lapply(dataSets, function(x) {
+    .makeDataFrameColumn(x, "LLOQ")
+  }), use.names = FALSE)
 
   df <- data.frame(
-    name, xValue, yValue, yErrorValues, xDimension, xUnit, yDimension,
-    yUnit, yErrorType, yErrorUnit, yMolWeight
+    name, xValues, yValues, yErrorValues, xDimension, xUnit, yDimension,

```

Comment:
Renamed column names (using plural) and added a LLOQ column.

Reply:
if feels like you could write a method
```
makeColums <- function(dataSets, columnName) {
  unlist(lapply(dataSet, function(x){
.makeDatqaFrameColumn(x, columnsName)
}
}

```
or sthg like this